akka-remote/src/main/scala/akka/remote/artery/tcp/ArteryTcpTransport.scala

```diff
@@ -333,6 +340,8 @@ private[remote] class ArteryTcpTransport(
     }
 
     attachInboundStreamRestart("Inbound streams", completed, restart)
+
+    boundPort.future
```

Comment:
Seems like that promise is strictly completed right before?

Reply:
You're right. I guess the `Await` just seemed 'icky' and I didn't want it to 'seep into' the API - though since it's a private API I guess it's fine. I'll update